code/game/objects/items/weapons/RPD.dm

```diff
@@ -0,0 +1,181 @@
+<A href='?src=\ref[src];make=4;dir=1'>Connector</A><BR>
+<A href='?src=\ref[src];make=7;dir=1'>Unary Vent</A><BR>
+<A href='?src=\ref[src];make=9;dir=1'>Gas Pump</A><BR>
+<A href='?src=\ref[src];make=15;dir=1'>Pressure Regulator</A><BR>
+<A href='?src=\ref[src];make=16;dir=1'>High Power Gas Pump</A><BR>
+<A href='?src=\ref[src];make=10;dir=1'>Scrubber</A><BR>
+<A href='?src=\ref[src];makemeter=1;dir=0'>Meter</A><BR>
+<A href='?src=\ref[src];make=13;dir=1'>Gas Filter</A><BR>
+<A href='?src=\ref[src];make=23;dir=1'>Gas Filter - Mirrored</A><BR>
+<A href='?src=\ref[src];make=14;dir=1'>Gas Mixer</A><BR>
+<A href='?src=\ref[src];make=25;dir=1'>Gas Mixer - Mirrored</A><BR>
+<A href='?src=\ref[src];make=24;dir=1'>Gas Mixer - T</A><BR>
+<A href='?src=\ref[src];make=26;dir=1'>Omni Gas Mixer</A><BR>
+<A href='?src=\ref[src];make=27;dir=1'>Omni Gas Filter</A><BR>
+<b>Heat exchange:</b><BR>
+<A href='?src=\ref[src];make=2;dir=1'>Pipe</A><BR>
+<A href='?src=\ref[src];make=3;dir=5'>Bent Pipe</A><BR>
+<A href='?src=\ref[src];make=6;dir=1'>Junction</A><BR>
+<A href='?src=\ref[src];make=17;dir=1'>Heat Exchanger</A><BR>
+<b>Insulated pipes:</b><BR>
+<A href='?src=\ref[src];make=11;dir=1'>Pipe</A><BR>
+<A href='?src=\ref[src];make=12;dir=5'>Bent Pipe</A><BR>
+
+"}
+///// Z-Level stuff
+//What number the make points to is in the define # at the top of construction.dm in same folder
+
+	user << browse("<HEAD><TITLE>[src]</TITLE></HEAD><TT>[dat]</TT>", "window=pipedispenser")
+	onclose(user, "pipedispenser")
+	return
+
+/obj/item/weapon/rpd/Topic(href, href_list)
+
+	if(..())
+		return
+	if(href_list["make"])
+		pipe_type = text2num(href_list["make"])
+		p_dir = text2num(href_list["dir"])
+	if(href_list["makemeter"])
+		p_dir = text2num(href_list["dir"]) //If direction is 0, the pipe is pipe_meter.
+	return
+
+/obj/item/weapon/rpd/afterattack(atom/A, mob/user, proximity)
+	if(wait)
+	//So that charge isn't used while wait is true. Doesnae solve the charge use bug, but does partially solve it.
+		return 0
+	if(!proximity)
+		return
+	if(!can_use(user,A))
+		return 0
+	if(istype(get_area(A),/area/shuttle)||istype(get_area(A),/turf/space/transit))
+		return 0
+	if(!useCharge(user))
+		to_chat(user, SPAN_WARNING("[src] battery is dead or missing."))
+		return 0
+	if(deletePipe(A))
+		to_chat(user, SPAN_NOTICE("You put [A] back to [src]."))
+		return
+	if(!p_dir == 0) //If direction is 0, the pipe is pipe_meter, otherwise it's one of the pipes.
+		if(!wait)
+			var/obj/item/pipe/P = new (A, pipe_type=pipe_type, dir=p_dir)
+			P.update()
+			P.add_fingerprint(usr)
+			wait = 1
+			spawn(10)
+				wait = 0
+	if(!wait)
+		new /obj/item/pipe_meter(A)
+		wait = 1
+		spawn(15)
+			wait = 0
+	return
+
+/obj/item/weapon/rpd/borg
+
+/obj/item/weapon/rpd/borg/useCharge(mob/user)
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(R.cell)
+			var/cost = use_power_cost*30 //Redundant values.
+			if(R.cell.charge >= cost)
```

Comment:
Change this and 173 to if(R.cell.checked_use(cost))

Reply:
`/obj/item/weapon/rpd/borg/useCharge(mob/user)
	if(isrobot(user))
		var/mob/living/silicon/robot/R = user
		if(R.cell)
			var/cost = use_power_cost*30 //Redundant values.
			if(R.cell.checked_use(cost))
				return
			return 1
	return 0`
So this?